front50-core/src/test/groovy/com/netflix/spinnaker/front50/model/pipeline/PipelineSpec.groovy

```diff
@@ -0,0 +1,60 @@
+package com.netflix.spinnaker.front50.model.pipeline
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.front50.api.model.pipeline.Pipeline
+import com.netflix.spinnaker.front50.api.model.pipeline.Trigger
+import com.netflix.spinnaker.front50.jackson.TriggerDeserializer
+import com.netflix.spinnaker.front50.jackson.mixins.PipelineMixins
+import spock.lang.Specification
+
+class PipelineSpec extends Specification {
+  ObjectMapper objectMapper = new ObjectMapper()
+
+  void setup() {
+    objectMapper.addMixIn(Pipeline.class, PipelineMixins.class)
+  }
+
+  def 'should ignore correct pipeline properties from deserializing Pipeline object to JSON'() {
+    given:
+    objectMapper.addMixIn(Pipeline.class, PipelineMixins.class)
+    Pipeline pipeline = new Pipeline()
+    pipeline.setId("1")
+    String pipelineJSON = objectMapper.writeValueAsString(pipeline)
+
+    expect:
+    pipelineJSON == "{}"
+  }
+
+  def 'should ignore correct pipeline properties from serializing JSON to Pipeline object'() {
+    given:
+    objectMapper.addMixIn(Pipeline.class, PipelineMixins.class)
+    String pipelineJSON = "{\"type\": \"fly\"}"
+
+    Pipeline pipeline = objectMapper.readValue(pipelineJSON, Pipeline.class)
+
+    expect:
+    pipeline.getType() == null
+  }
+
+  def 'should grab triggers after deserializing JSON into Pipeline objet'() {
+    given:
+    objectMapper.addMixIn(Pipeline.class, PipelineMixins.class)
+    String pipelineJSON = '{"triggers": [{"type": "cron", "id": "a"}, {"type": "cron", "id": "b"}]}'
+
+    ArrayList<Trigger> triggers = new ArrayList<Trigger>();
+    Trigger triggerA = new Trigger();
+    triggerA.put("type", "cron");
+    triggerA.put("id", "a");
+    triggers.add(triggerA);
+
+    Trigger triggerB = new Trigger();
+    triggerB.put("type", "cron");
+    triggerB.put("id", "b");
+    triggers.add(triggerB);
+
+    Pipeline pipeline = objectMapper.readValue(pipelineJSON, Pipeline.class)
+
+    expect:
+    TriggerDeserializer.deserialize(pipeline) == triggers
```

Comment:
Continuing on my comments above - shouldn't this work without the `TriggerDeserializer`?

Reply:
Just double checked - it does pass.